### ISSUE TITLE
Using an 'artifact' repo results in bad links and bad dist files

### ISSUE THREAD

**Issue by GaryRogers** [composer/satis#148]
Using an artifact repository results in bad links, and misnamed files in the dist directory.

``` json
{
    "name": "My Repository",
    "homepage": "http://localhost/artifacts",
    "repositories": [
        { "type": "artifact", "url": "./packages" }
    ],
        "archive": {
            "directory": "dist",
            "format": "tar"
        },  
    "require-all": true
}
```

creates links like: `http://localhost/artifacts/dist/local-mypackage-1.0.5.7-local-mypackage-1.0.5.7.zip.tar` as well as the corresponding `.zip.tar` files in `dist`

**Comment by alcohol** [comment]
Is this still an issue?

**Comment by leonboot** [comment]
The problem lies in the fact that the artifact repository's `url` parameter, which represents the path the artifacts reside in on the filesystem, is being used to form the package URL. Since the artifacts need to be exposed in the webroot (ie. the output dir for the `satis build` command), there's a mismatch between the physical location on the filesystem and the relative path to the package from Satis' web root.

I've tagged this issue in a commit on a fork I made of this repo. The fix is by far complete; I'm new to Satis and this is what I've been able to come up with after an hour or so trying to reverse engineer the way it works. Maybe the commit illustrates the problem and someone's able to come up with a more robust fix.

**Comment by jtanmay-dev** [comment]
is this merged? facing the same issue!

**Comment by alcohol** [comment]
Merged? This is an issue, not a pull-request.

**Comment by jtanmay-dev** [comment]
Yep. I thought @leonboot would have already sent the pull request, as he fixed it over a year ago! :)

**Comment by leonboot** [comment]
@jtanmay-dev I haven't filed a pull request, since I'm not sure the fix is complete or even correct. I haven't studied the innards of Satis well enough, so I thought I'd share what I did as a pointer for someone more qualified. But given the fact it's been over a year ago, I might have to consider having a second go at it and improve upon the fix and file a PR.

**Comment by forsvunnet** [comment]
Here's a workaround to use the fix @leonboot made:

```
git remote add nedbase https://github.com/nedbase/satis.git
git pull all
git cherry-pick fb54ffc2b5
```

**Comment by poldixd** [comment]
I created a sh script that replaces the relative url with an absolute url after generating the package lists

```
cd satisfy && /opt/plesk/php/7.1/bin/php bin/satis build satis.json web \;
cd web/include && find . -type f -exec sed -i 's#"url": "packages/#"url": "https://packages.example.org/packages/#g' {} \;
```

**Comment by MarWestermann** [comment]
Hi I found a workaround without using fix or additional scripts

I created a folder /data where I put module zip files

I have the following satis configuration:

`{
    "name": "My Composer Repo",
    "homepage": "http://satis.mydomain.com/" ,
    "repositories": [
        {
          "type": "artifact",
          "url": "/data"
        }
    ],
    "archive": {
        "directory": "data",
        "format": "zip",
        "prefix-url": "http:/satis.mydomain.com"
    },
    "require-all":true,
    "require-dependencies":true,
    "require-dev-dependencies":true,
    "minimum-stability":"dev"
}`

This way satis finds the zips and indexes them.
But at this point you are not able to download a zip file.

You also have to add a symlink: ln -s /data/ /satisfy/web/

Now everything works.
